bokehjs/src/coffee/embed.ts

```diff
@@ -88,8 +105,25 @@ function _init_comms(target: string, doc: Document): void {
     } catch (e) {
       logger.warn(`Jupyter comms failed to register. push_notebook() will not function. (exception reported: ${e})`)
     }
-  } else
+  } else if (doc.roots()[0].id in kernels) {
+    logger.info(`Registering JupyterLab comms for target ${target}`)
+    const kernel = kernels[doc.roots()[0].id]
+    for (const id in kernel._comms) {
+      const promise = kernel._comms[id]
+      promise.then(update_comms)
+    }
+    try {
+      kernel.registerCommTarget(target, (comm: Comm) => {
+        logger.info(`Registering JupyterLab comms for target ${target}`)
+        const r = new Receiver()
+        comm.onMsg = _handle_notebook_comms.bind(doc, r)
+      })
+    } catch (e) {
+      logger.warn(`Jupyter comms failed to register. push_notebook() will not function. (exception reported: ${e})`)
+    }
+  } else {
     console.warn('Jupyter notebooks comms not available. push_notebook() will not function');
```

Comment:
What would lead to this branch? (Can the error message be improved)

Reply:
This branch would be hit if you're exporting the HTML, or if you haven't installed the jupyterlab extension. No way of knowing which really.

Reply:
OK, I'd suggest mentioning those possibilities either in the message or in some docs somewhere.

Reply:
Will do.